src/lib.rs

```diff
@@ -2467,6 +2450,54 @@ impl Build {
             println!("{}", s);
         }
     }
+
+    #[cfg(target_os = "macos")]
+    fn fix_env_for_apple_os(&self, cmd: &mut Command) -> Result<(), Error> {
+        let target = self.get_target()?;
+        let host = self.get_host()?;
+        if host.contains("apple-darwin") && target.contains("apple-darwin") {
+            // Replace the `SDKROOT` environment variable if it's clearly set for the wrong platform, which
+            // may occur when we're linking a custom build script while targeting iOS for example.
+            // Removing is not enough, because of cc from XCode can not find include files in such case
```

Comment:
Can this comment be clarified? The point of "removing is not enough" is only applicable when we're running `clang` directly out of the sysroot I suspect. Executing `/usr/bin/clang` doesn't require `SDKROOT` to be set, for example.

Reply:
Done